mongo/client.go

```diff
@@ -73,6 +74,7 @@ type Client struct {
 	timeout        *time.Duration
 	httpClient     *http.Client
 	logger         *logger.Logger
+	closed         atomic.Value
```

Comment:
We don't need to track this, instead we should rely on the [state of the topology](https://github.com/mongodb/mongo-go-driver/blob/92c3d6e5425fe7b9b3b5c70b18683d9c7c5ffc98/mongo/errors.go#L64). I suspect the reason [this](https://github.com/mongodb/mongo-go-driver/pull/1807/files#diff-005798954aa04ac42c733486ab348b122d60437c462aae4665c90e4d66448cc4L81) test is failing is because of the change to [setupClient](https://github.com/mongodb/mongo-go-driver/pull/1807/files#diff-4220cfa18a0cd4b8c2753f5a72e2f1f6645593197b897033c3631809eb888d13L40) which no longer makes it possible to create a client with a null session pool. To resolve this, I suggest disconnecting the topology directly:

```go
topo, ok := coll.client.deployment.(*topology.Topology)
require.True(t, ok, "client deployment is not a topology")

err := topo.Disconnect(context.Background())
require.NoError(t, err)
```